04/app.js

```diff
@@ -5,18 +5,43 @@ import File from './File';
 import List from './List';
 
 class App extends React.Component {
-    state = {
-        filesList: [],
-    }
+  state = {
+    filesList: [],
+  };
 
-    render() {
-        return (
-            <section>
-                <File />
-                <List />
-            </section>
-        )
-    }
+  saveFileIntoState = () => {
+    const uploadedFiles = this.fileReader.uploadedFiles();
+
+    uploadedFiles.map((file) => {
+      if (file && file.type.includes('text')) {
+        const reader = new FileReader();
+        reader.readAsText(file, 'UTF-8');
+        reader.onload = () => {
+          const fileToPushIntoState = {
+            name: file.name,
+            content: reader.result,
+            size: file.size,
+          };
+
+          this.setState({
+            filesList: [...this.state.filesList, fileToPushIntoState],
+          });
+        };
+      }
+    });
+  };
+
+  render() {
+    return (
+      <section>
+        <File
+          ref={(element) => (this.fileReader = element)}
```

Comment:
Tutaj również zastanawiam się czy nie wygodnie byłoby użyć dodatkowego `state`.

Reply:
`W komponencie <App /> mamy state, w którym będziemy przechowywać informacje o zawartości pliku, jego nazwie oraz rozmiarze.`
Czy chodzi o to, by utworzyć state w komponencie File? :) Jeśli tak, to do czego mógłby służyć?